geode-core/src/main/java/org/apache/geode/management/internal/cli/domain/RegionConfigFactory.java

```diff
@@ -15,44 +15,84 @@
 package org.apache.geode.management.internal.cli.domain;
 
 import java.util.Optional;
+import java.util.Set;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
+import org.apache.geode.cache.CacheListener;
+import org.apache.geode.cache.CacheLoader;
+import org.apache.geode.cache.CacheWriter;
+import org.apache.geode.cache.CustomExpiry;
+import org.apache.geode.cache.ExpirationAction;
 import org.apache.geode.cache.RegionAttributes;
 import org.apache.geode.cache.configuration.ClassNameType;
 import org.apache.geode.cache.configuration.DeclarableType;
+import org.apache.geode.cache.configuration.EnumActionDestroyOverflow;
 import org.apache.geode.cache.configuration.ExpirationAttributesType;
 import org.apache.geode.cache.configuration.RegionAttributesScope;
 import org.apache.geode.cache.configuration.RegionAttributesType;
 import org.apache.geode.cache.configuration.RegionConfig;
-import org.apache.geode.management.internal.cli.functions.RegionFunctionArgs;
 
 public class RegionConfigFactory {
-  public RegionConfig generate(RegionFunctionArgs args) {
+  public RegionConfig generate(
```

Comment:
I wonder if this is really worthwhile. This function is going to be so hard to maintain and test. I thought we would completely get rid of this factory classes, and just construct the config object right in the command and set the values as we validate each group of parameters. We do not want to repeat the same number parameters as the command again. It's easy to unit test the command class using the parserRule, but it's not easy to unit test this.

Reply:
The reason we had the factory class in the first place is to be able to test-drive in isolation. The original unit test was quite easy, it's just that with all these parameters it's a little harder to read. But I'm not sure if I agree that it would be easier if this was all in CreateRegionCommand?

Reply:
but having this huge list of parameters in method is very anti-pattern. You can use Junit's ArgumentCaptor to capture the RegionConfig passed to the function call and verify that way.

Reply:
Paired on this with @jinmeiliao, we have a resolution.